### PR TITLE
Feature #1272: Window Distinct Merging

### PR DESCRIPTION
**Pull request by hawkfish**
Fully parallelise the sorting for distinct aggregates.

### CODE REVIEW

**Comment by hawkfish** [comment]
> Slowly, slowly, catchy monkey
> - _Killing Eve_, S3:E1

File: src/execution/window_segment_tree.cpp [hunk 1]
```diff
@@ -1544,13 +1566,87 @@ void WindowDistinctAggregatorLocalState::Sink(DataChunk &arg_chunk, idx_t input_
...
+	lock_guard<mutex> stage_guard(lock);
+
+	switch (stage) {
+	case PartitionSortStage::INIT:
+		// Wait for all the local sorts to be processed
+		if (tasks_completed < locals) {
+			return false;
+		}
+		global_sort->PrepareMergePhase();
+		total_tasks = global_sort->sorted_blocks.size() / 2;
```
Comment (lnkuiper) [inline src/execution/window_segment_tree.cpp:1598]: Won't the parallelism slowly go down to one thread, as you only have one task if there's one sorted block?

We can merge two sorted blocks using many threads. Would it be better to take a morsel-driven approach here and parallellize based on the size of the data, i.e., number of rows?

**Review by lnkuiper — COMMENTED** [review]
Thanks for the PR! I just have one question:

File: src/execution/window_segment_tree.cpp (hunk 1 above)
Comment (hawkfish) [reply src/execution/window_segment_tree.cpp]: Because I didn't realise it did that!

**Review by lnkuiper — APPROVED** [review]
Looks good to me now. Thanks!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**